src/TextDateInput/TextDateInput.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { useEffect, useState } from 'react'
+import styled from 'styled-components'
+
+import { TextInput } from '../TextInput'
+import { Box } from '../Box'
+import { Fieldset, FieldsetProps } from '../fields/Fieldset'
+import { dateOfBirthValidator } from '../utils/dateOfBirth/dateOfBirthValidator'
+import { createErrorMessage } from '../utils/dateOfBirth/schema'
+
+export interface DateObjectType {
+  day: string
+  month: string
+  year: string
+}
+
+export type DateObject = {
+  [K in keyof DateObjectType]?: DateObjectType[K] | null
+}
+
+export type TextDateInputProps = {
+  value: DateObjectType
+  onChange: (value: DateObjectType) => void
+  showCompleted?: boolean
+} & Pick<FieldsetProps, 'label' | 'error' | 'assistiveText'>
+
+export const TextDateInput = ({
+  onChange,
+  value,
+  label,
+  assistiveText,
+  error,
+  showCompleted = false,
+}: TextDateInputProps) => {
+  const [errorMsg, setErrorMsg] = useState<string | undefined>(undefined)
+  const [fieldsWithError, setFieldsWithError] = useState<string[]>([])
+
+  const handleValidationErrors = () => {
+    const validationResult = dateOfBirthValidator(value)
```

Comment:
I would argue, because of this validator, this component isn't a generic `TextDateInput` but a more opinionated `DateOfBirthDateInput` component.

I think we'll need to decide how to handle configurable validation before this component can be generic enough to be named `TextDateInput`.

Example:
- One would assume this component is usable for all past dates but the validator doesn't allow for dates less than 17 years ago
- One would assume this component is usable for future dates but doesn't